### PR TITLE
Update student-t sampling to use correct key for gamma

### PR DESCRIPTION
**Pull request by Tennessee-Wallaceh**
Fix for issue #14597. 
Simple change to use correct random key in sampling procedure.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks good! Let's run the tests to see if anything needs fixing.

**Comment by jakevdp** [comment]
It looks like the test of the statistical properties of the distribution is failing... that test is notoriously finnicky, because it is only probabilistically correct. You may be able to fix it by adjusting the test's random seed or number of generated elements

**Review by jakevdp — COMMENTED** [review]
Looks good, thanks! As a last step, could you please squash your changes into a single commit? (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests) thanks!
